homeassistant/components/sensor/solaredge.py

```diff
@@ -0,0 +1,161 @@
+
+REQUIREMENTS = ['solaredge==0.0.2']
+
+DOMAIN = "solaredge"
+
+# Config for solaredge monitoring api requests.
+CONF_API_KEY = "api_key"
+CONF_SITE_ID = "site_id"
+
+DELAY_OK = 10
+DELAY_NOT_OK = 20
+
+# Supported sensor types:
+# Key: ['name', unit, icon]
+SENSOR_TYPES = {
+    'lifeTimeData': ["Lifetime energy", 'Wh', 'mdi:solar-power'],
+    'lastYearData': ["Energy this year", 'Wh', 'mdi:solar-power'],
+    'lastMonthData': ["Energy this month", 'Wh', 'mdi:solar-power'],
+    'lastDayData': ["Energy today", 'Wh', 'mdi:solar-power'],
+    'currentPower': ["Current Power", 'W', 'mdi:solar-power']
+}
+
+PLATFORM_SCHEMA = PLATFORM_SCHEMA.extend({
+    vol.Required(CONF_API_KEY): cv.string,
+    vol.Required(CONF_SITE_ID): cv.string,
+    vol.Optional(CONF_MONITORED_VARIABLES, default=[]):
+    vol.All(cv.ensure_list, [vol.In(SENSOR_TYPES)])
+})
+
+_LOGGER = logging.getLogger(__name__)
+
+
+@asyncio.coroutine
+def async_setup_platform(hass, config, async_add_entities,
+                         discovery_info=None):
+    """Create the SolarEdge Monitoring API sensor."""
+    api_key = config.get(CONF_API_KEY, None)
+    site_id = config.get(CONF_SITE_ID, None)
+
+    # Create solaredge data service which will retrieve and update the data.
+    data = SolarEdgeData(hass, api_key, site_id)
+
+    # Create a new sensor for each sensor type.
+    entities = []
+    for sensor_type in config[CONF_MONITORED_VARIABLES]:
+        sensor = SolarEdgeSensor(sensor_type, data)
+        entities.append(sensor)
+
+    async_add_entities(entities, True)
+
+    # Schedule first data service update straight away.
+    async_track_point_in_utc_time(hass, data.async_update, dt_util.utcnow())
+
+
+class SolarEdgeSensor(Entity):
+    """Representation of an SolarEdge Monitoring API sensor."""
+
+    def __init__(self, sensorType, data):
+        """Initialize the sensor."""
+        self.type = sensorType
+        self.data = data
+
+        self._name = SENSOR_TYPES[self.type][0]
+        self._unit_of_measurement = SENSOR_TYPES[self.type][1]
+
+    @property
+    def name(self):
+        """Return the name."""
+        return 'solaredge_' + self.type
+
+    @property
+    def unit_of_measurement(self):
+        """Return the unit of measurement."""
+        return self._unit_of_measurement
+
+    @property
+    def icon(self):
+        """Return the sensor icon."""
+        return SENSOR_TYPES[self.type][2]
+
+    @property
+    def state(self):
+        """Return the state of the sensor."""
+        if self.type in self.data.data:
+            return self.data.data.get(self.type)
+        return 0
+
+
+class SolarEdgeData:
+    """Get and update the latest data."""
+
+    def __init__(self, hass, api_key, site_id):
+        """Initialize the data object."""
+        self.hass = hass
+        self.data = {}
+        self.api_key = api_key
+        self.site_id = site_id
+
+    @asyncio.coroutine
+    def schedule_update(self, minutes):
```

Comment:
Remove this and add a throttle to `async_update()` and perhaps introduce `available()` if the API is unstable.